services/src/main/java/org/keycloak/exportimport/util/ExportUtils.java

```diff
@@ -107,7 +107,6 @@ public static RealmRepresentation exportRealm(KeycloakSession session, RealmMode
 
         if (options.isClientsIncluded()) {
             clients = realm.getClientsStream()
-              .filter(c -> { try { c.getClientId(); return true; } catch (Exception ex) { return false; } } )
```

Comment:
This filtering should be kept by using `ModelToRepresentation.filterValidRepresentations(realm.getClientsStream(), app -> exportClient(session, app)` below.

Reply:
Hi @hmlnarik, I did this change intentionally as stated above, see my comment: 

> ExportUtils will fail if a client can't be exported ... this changes the existing behavior, but seems to be reasonable as exportutils should export everything, and no clients should be missing.

I now change the code as you requested. As the code needs a list of ClientModels later to process groups etc., the code is now a bit more complex to construct the list of client models that are valid client representations. Please let me know what you think.

Reply:
A client might be missing permanently, leading to permanently thrown exception, and throwing an exception effectively removes ability to run export; so I'd rather keep it as is with your current changes.

Reply:
I could see a situation where someone would be happy to know about this problem by seeing the exception, and I also see the case you make here. 

When you're happy with the change, then please resolve this thread.